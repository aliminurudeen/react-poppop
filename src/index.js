import PopPop from './PopPop';

export default PopPop;