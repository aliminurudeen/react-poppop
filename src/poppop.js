import React, {Component} from 'react';
import STYLE from './style';

export default class PopPop extends Component {

  constructor(props) {
    super(props);
    this.state = {
      display: this.props.display
    }

    this.handleOverlayClick = this.handleOverlayClick.bind(this);
    this.hide = this.hide.bind(this);
  }

  handleOverlayClick() {
    const {
      overlayClick
    } = this.props;

    if (overlayClick) {
      this.hide();
    }
  }

  show() {
    this.setState({display: 'show'});
  }

  hide() {
    this.setState({display: 'hide'});
  }

  render() {
    let overlayTmpl,
        displayStyle = {};
    const {
      overlay,
      position
    } = this.props;
    const display = this.state.display;

    if (display === 'hide') {
      displayStyle.display = 'none';
    } else if (display === 'show') {
      displayStyle.display = 'block';
    }

    // merge the content style and position style
    const wrapperStyle = Object.assign(STYLE.wrapper, STYLE[position]);
    const contentStyle = Object.assign(STYLE.content, displayStyle);
    const overlayStyle = Object.assign(STYLE.overlay, displayStyle);

    if (overlay) {
      overlayTmpl = <div style={STYLE.overlay} 
                          onClick={this.handleOverlayClick}></div>;
    }

    return (
      <div style={wrapperStyle}> 
        {overlayTmpl}
        <div style={contentStyle}>
          <button style={STYLE.closeBtn}
                  onClick={this.hide}>x</button>
          {this.props.children}
        </div>
      </div>
    )
  }
}
