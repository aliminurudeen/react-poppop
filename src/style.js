module.exports = {

  wrapper: {
    opacity: '1',
    visibility: 'visible',
    position: 'fixed',
    overflow: 'auto',
    zIndex: '100001',
    transition: 'all 0.3s',
    width: '100%',
    height: '100%',
    top: '0px',
    left: '0px',
    display: 'flex',
    alignItems: 'center'
  },

  content: {
    transition: 'all 0.3s',
    backgroundColor: 'white',
    borderRadius: '3px',
    zIndex: 100,
    position: 'relative',
    boxShadow: '0 0 4px rgba(0,0,0,.14),0 4px 8px rgba(0,0,0,.28)'
  },

  center: {
    verticalAlign: 'middle',
    justifyContent: 'center'
  },

  left_center: {
    verticalAlign: 'middle',
    justifyContent: 'flex-start'
  },

  right_center: {
    verticalAlign: 'middle',
    justifyContent: 'flex-end'
  },

  full: {
    width: '100%',
    height: '100%'
  },

  overlay : {
    position: 'fixed',
    top: 0,
    left: 0,
    width: '100%',
    height: '100%',
    zIndex: 99,
    backgroundColor: 'rgba(0,0,0,0.3)'
  },

  closeBtn: {
    position: 'absolute',
    right: '-15px',
    top: '-15px',
    backgroundColor: 'black',
    color: 'white',
    borderRadius: '50%',
    boxShadow: '0 0 10px 0 #000',
    cursor: 'pointer',
    width:'30px',
    height:'30px',
    lineHeight: '34px',
    textAlign: 'center',
    fontSize: '13px'
  },

  fullCloseBtn: {
    right: '0',
    top: '0',
    borderRadius: '0'
  }
}