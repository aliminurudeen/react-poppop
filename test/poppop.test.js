import React from 'react';
import PopPop from '../src';
import {shallow, render, mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import {extractCamelCase} from '../src/utils';
import {alignItems, justifyContent} from '../src/style';

const shareComponent = props => (
  <PopPop open={true} {...props}>
    content
  </PopPop>
);

describe('render', () => {
  it('no props', () => {
    const component = mount(shareComponent());
    expect(toJson(component)).toMatchSnapshot();
  });

  it('close', () => {
    const component = mount(shareComponent({open: false}));
    expect(component.html()).toEqual(null);
  })
})

describe('props', () => {
  it('position style', () => {
    const positionList = [
      'topLeft', 'topCenter', 'topRight',
      'centerLeft', 'centerCenter', 'centerRight',
      'bottomLeft', 'bottomCenter', 'bottomRight',
    ];
    positionList.forEach(position => {
      const component = mount(shareComponent({position}))
      const extract = extractCamelCase(position);

      expect(component.find('div').get(0).props.style).toMatchObject(alignItems[extract[0]])
      expect(component.find('div').get(0).props.style).toMatchObject(justifyContent[extract[1]])
    })
  })

  it('overlay style', () => {
    const component = mount(shareComponent({overlayStyle: {background: 'red'}}));
    expect(component.find('Tappable').get(0).props.style).toMatchObject({background: 'red'});
  })

  it('content style', () => {
    const component = mount(shareComponent({contentStyle: {background: 'red'}}));
    expect(component.find('div').get(1).props.style).toMatchObject({background: 'red'});
  })

  it('has close button & trigger onClose', () => {
    const mockClose = jest.fn();
    const component = mount(shareComponent({onClose: mockClose, closeBtn: true}));
    component.find('svg').at(0).simulate('click');
    expect(mockClose).toBeCalled();
  })
})
