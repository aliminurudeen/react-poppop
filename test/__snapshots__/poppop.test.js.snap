// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render no props 1`] = `
<PopPop
  closeOnOverlay={true}
  open={true}
  overlayStyle={Object {}}
  position="topCenter"
>
  <Portal>
    <Transition
      appear={true}
      enter={true}
      exit={true}
      in={true}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={0}
      unmountOnExit={false}
    >
      <div
        style={
          Object {
            "alignItems": "flex-start",
            "bottom": "0",
            "display": "flex",
            "justifyContent": "center",
            "left": "0",
            "opacity": 0,
            "overflowX": "hidden",
            "overflowY": "auto",
            "padding": "20px",
            "position": "fixed",
            "right": "0",
            "top": "0",
            "transition": "opacity .2s linear",
            "visibility": "visible",
            "zIndex": "1000",
          }
        }
      >
        <div
          onClick={[Function]}
          onTouchTap={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(0,0,0,0.6)",
              "height": "100%",
              "left": 0,
              "position": "fixed",
              "top": 0,
              "width": "100%",
              "zIndex": 99,
            }
          }
        />
        <div
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": "3px",
              "boxShadow": "0 0 4px rgba(0,0,0,.14),0 4px 8px rgba(0,0,0,.28)",
              "maxWidth": "850px",
              "overflow": "scroll",
              "padding": "10px 20px",
              "position": "relative",
              "transition": "all 0.3s",
              "zIndex": 2000,
            }
          }
        >
          content
        </div>
      </div>
    </Transition>
  </Portal>
</PopPop>
`;
